File: two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(const vector<int>& nums, const int& target) {
        
        vector<int> sum;
        
        // value, index
        unordered_map<int, int> uMap;
        
        for (int i = 0; i < nums.size(); ++i) {
            
            int difference = target - nums[i];
            
            if (uMap.find(difference) != uMap.end()) {
                
                sum.push_back(uMap[difference]);
                sum.push_back(i);                
                
                return sum;
            }
            
            uMap[nums[i]] = i;
        }
        
        return sum;
    }
};

File: two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int[] sum = {0, 0};
        
        HashMap<Integer, Integer> hMap = new HashMap<>();
        
        for (int i = 0; i < nums.length; ++i) {
            
            int difference = target - nums[i];
            
            if (hMap.containsKey(difference)) {
                
                sum[0] = hMap.get(difference);
                sum[1] = i;
                
                return sum;
            }
            
            hMap.put(nums[i], i);
        } 
        
        return sum;
    }
}